motionplan/collision.go

```diff
@@ -242,10 +242,22 @@ func newCollisionGraph(x, y []spatial.Geometry, reference *collisionGraph, repor
 // checkCollision takes a pair of geometries and returns the distance between them.
 // If this number is less than the CollisionBuffer they can be considered to be in collision.
 func (cg *collisionGraph) checkCollision(x, y spatial.Geometry) (float64, error) {
+	// x is the robot geometries and therefore must be primitives in spatialmath
+	// y could potentially be a geometry type that lives outside spatialmath and therefore knows more so we defer to it for collisions
```

Comment:
[q] how is y 'potentially' a geometry type when it is type cast as one in the function def?

Reply:
the 'potentially' is in reference to the geometry maybe living outside spatialmath (basically y can be an octree which is outside spatialmath). I can make the wording clearer

Reply:
That would be helpful thanks!

Reply:
I actually think we can just remove this comment as I think it causes more confusion than it clears up.  Now that we're trying both ways we aren't doing something super hacky that warrants explanation